README.md

```diff
@@ -6,63 +6,139 @@
 
+## Secrets & credentials
+This module presents two ways of setting credentials (username and password). You can set them manually or upload secrets to Vault.
+
+### Set credentials manually
+This module uses the `admin_user` and `admin_password` credentials as default, and contain the value `postgres`.  
+If you want to overwrite the default values, you need to is set the `vault_secret.use_vault_kv` to `false` and leave the rest of the values as blanks. 
+Finally, you can change the default username and password. 
+E.g:
+```hcl-terraform
+module "postgres" {
+...
+
+  vault_secret  = {
+                    use_vault_kv           = false,
+                    vault_kv_path          = "",
+                    vault_kv_username_name = "",
+                    vault_kv_password_name = ""
+                  }
+  admin_user     = "myadminuser"     # default 'postgres'
+  admin_password = "myadminpassword" # default 'postgres'
+}
+```
+
+### Set credentials using Vault secrets
+By default the Vault secrets are not activated. In other words the `vault_secret.use_vault_kv` is set to `false`. 
+However, when testing using the box (e.g. `make dev`) the postgres username and password is randomly generated and put in `secret/postgres` inside Vault, from the [01_generate_secrets_vault.yml](dev/ansible/01_generate_secrets_vault.yml) playbook. 
+This is a independet process and will run regardless of the `vault_secret.use_vault_kv` is `false/true`. 
+
+If you want to use the automatically generated credentials in the box, you can do so by changing the `vault_secret` objcet as seen bellow:
+```hcl-terraform
+module "postgres" {
+...
+
+  vault_secret  = {
+                    use_vault_kv           = true,
+                    vault_kv_path          = "secret/postgres",
+                    vault_kv_username_name = "username",
+                    vault_kv_password_name = "password"
+                  }
+}
+```
+
+If you where to change the secrets path and keys/values in Vault with your own configuration you would need to change the variables in the `vault_secret`-object. 
+Say that you have put your secrets in `secret/services/postgres/users` and change the keys to `guestuser` and `guestpassword`. Then you need to do the following configuration:
+```hcl-terraform
+module "postgres" {
+...
+
+  vault_secret  = {
+                    use_vault_kv           = true,
+                    vault_kv_path          = "secret/services/postgres/users",
+                    vault_kv_username_name = "guestuser",
+                    vault_kv_password_name = "guestpassword"
+                  }
+}
+```
+
+## Volumes
+We are using [host volume](https://www.nomadproject.io/docs/job-specification/volume) to store postgres data.
+If the `use_host_volume` is set to `true` (default: false), Postgres data will be available in root `/persistence/postgres` folder inside the Vagrant box.
+
 ## Example usage
 The example-code shows the minimum of what you need to use this module.
```

Comment:
Technically none of the variables are needed as a minimum, because they have defaults. But that might not be as instructive in learning how to use the module maybe?

What about removing all variables from the minimum example, and provide a link to where you can see the defaults + an example on how to override them

Reply:
I'll rephrase it!:+1: